shoop/admin/modules/suppliers/views/edit.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shoop.
+#
+# Copyright (c) 2012-2015, Shoop Ltd. All rights reserved.
+#
+# This source code is licensed under the AGPLv3 license found in the
+# LICENSE file in the root directory of this source tree.
+from __future__ import unicode_literals
+
+from django import forms
+from django.utils.translation import ugettext_lazy as _
+
+from shoop.admin.utils.views import CreateOrUpdateView
+from shoop.core.models import Supplier
+
+
+class SupplierForm(forms.ModelForm):
+    class Meta:
+        model = Supplier
+        exclude = ("module_data",)
+        widgets = {
+            "module_identifier": forms.Select(
+                choices=model.get_module_choices(
```

Comment:
Mmmmm... `get_module_choices` can't necessarily be safely called at import time (it may `load()` something with circular dependencies).

Move the choices assignment to `get_form()` in the view instead, please :) It doesn't need to be anything more complex than

```python
def get_form(self, form_class=None):
    form = super(SupplierEditView, self).get_form(form_class=form_class)
    choices = self.model.get_module_choices(...)
    form["module_identifier"].choices = form["module_identifier"].widget.choices = choices
    return form
````